File: example/android/app/src/main/java/io/intheloup/streamschannelexample/DemoPluginRegistrant.java
//  Copyright (c) 2018 Loup Inc.
//  Licensed under Apache License v2.0

package io.intheloup.streamschannelexample;

import io.flutter.plugin.common.PluginRegistry;

public final class DemoPluginRegistrant {
  public static void registerWith(PluginRegistry registry) {
    if (alreadyRegisteredWith(registry)) {
      return;
    }
    DemoPlugin.registerWith(registry.registrarFor("io.intheloup.streamschannelexample.DemoPlugin"));
  }

  private static boolean alreadyRegisteredWith(PluginRegistry registry) {
    final String key = DemoPluginRegistrant.class.getCanonicalName();
    if (registry.hasPlugin(key)) {
      return true;
    }
    registry.registrarFor(key);
    return false;
  }
}

File: example/ios/Runner/DemoPluginRegistrant.m
//  Copyright (c) 2018 Loup Inc.
//  Licensed under Apache License v2.0

#import "DemoPluginRegistrant.h"
#import "DemoPlugin.h"

@implementation DemoPluginRegistrant

+ (void)registerWithRegistry:(NSObject<FlutterPluginRegistry>*)registry {
  [DemoPlugin registerWithRegistrar:[registry registrarForPlugin:@"DemoPlugin"]];
}

@end
